pkg/providers/cloudstack/template.go

```diff
@@ -66,20 +50,14 @@ func getWorkerNodeGroupMachineSpec(clusterSpec v1alpha1.ClusterSpec, machineConf
 		}
 	}
 
-	if len(workerNodeGroupMachineSpecs) == 0 {
-		return workerNodeGroupMachineSpecs, errors.Errorf("getting worker node group configs")
-	}
-
-	return workerNodeGroupMachineSpecs, nil
+	return workerNodeGroupMachineSpecs
 }
 
-func getControlPlaneMachineSpec(clusterSpec v1alpha1.ClusterSpec, machineConfigs map[string]*v1alpha1.CloudStackMachineConfig) (*v1alpha1.CloudStackMachineConfigSpec, error) {
+func getControlPlaneMachineSpec(clusterSpec v1alpha1.ClusterSpec, machineConfigs map[string]*v1alpha1.CloudStackMachineConfig) *v1alpha1.CloudStackMachineConfigSpec {
 	var controlPlaneMachineSpec *v1alpha1.CloudStackMachineConfigSpec
 	if clusterSpec.ControlPlaneConfiguration.MachineGroupRef != nil && machineConfigs[clusterSpec.ControlPlaneConfiguration.MachineGroupRef.Name] != nil {
```

Comment:
Should we remove this first if case then since it'll never be nil at this point?

Reply:
I don't think this is necessary for now.

To clarify, the main reason I refactored this is to avoid the extra work of writing a test to cover these cases that would eventually be removed soon once the `CloudStackTemplateBuilder` has been simplified.

Reply:
ok sounds good!